FirebaseDatabase/Sources/Core/FRepoInfo.h

```diff
@@ -16,25 +16,48 @@
 
 #import <Foundation/Foundation.h>
 
+NS_ASSUME_NONNULL_BEGIN
+
 @interface FRepoInfo : NSObject <NSCopying>
 
-@property(nonatomic, readonly, strong) NSString *host;
-@property(nonatomic, readonly, strong) NSString *namespace;
-@property(nonatomic, strong) NSString *internalHost;
-@property(nonatomic, readonly) bool secure;
+/// Stores the original host if the host has been set after initialization.
+/// Used to validate references.
+@property(nonatomic, readwrite, copy, nullable) NSString *underlyingHost;
```

Comment:
I think you don't need this. If you do keep this, we may need some API wrapper so that we don't have `underlyingHost`, `host` and `internalHost` all in the same class.

Reply:
Removed.